spec/models/spree/user_decorator_spec.rb

```diff
@@ -19,8 +20,49 @@
     end
   end
 
+  describe '#supplier_admin?' do
+    it "returns true if user is a supplier and has admin role" do
+      user.supplier = build :supplier
+      user.spree_roles << admin_role
+
+      expect(user.supplier_admin?).to eq true
+    end
+
+    it "returns false if user is a supplier with no admin role" do
+      expect(user.supplier_admin?).to eq false
+    end
+
+    it "returns false if user is not a supplier but has admin role" do
+      user.spree_roles << admin_role
+      expect(user.supplier_admin?).to eq false
+    end
+  end
+
+  describe '#market_maker?' do
+    it "returns true if user doesn't have a supplier and has admin role" do
+      user.spree_roles << admin_role
+
+      expect(user.market_maker?).to eq true
+    end
+
+    it "returns true if user doesn't have a supplier and doesn't have an admin role" do
```

Comment:
the bodies of this test and the next are the same, but their descriptions are opposite